src/nupic/algorithms/KNNClassifier.py

```diff
@@ -406,6 +406,16 @@ def learn(self, inputPattern, inputCategory, partitionId=None, isSparse=0,
       print "  active inputs:", _labeledInput(inputPattern,
                                               cellsPerCol=self.cellsPerCol)
 
+    if isSparse > 0:
+      isSorted = all(inputPattern[i] <= inputPattern[i+1]
+                     for i in xrange(len(inputPattern)-1))
+      if not isSorted:
+        raise RuntimeError("Sparse inputPattern must be sorted.")
```

Comment:
Pick more specific exception types for these cases. Not sure what would be best for this one (`ValueError`?) but the next one could be `IndexError`.